staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/connection.go

```diff
@@ -217,6 +217,31 @@ func (connection *VSphereConnection) NewClient(ctx context.Context) (*vim25.Clie
 		connection.RoundTripperCount = RoundTripperDefaultCount
 	}
 	client.RoundTripper = vim25.Retry(client.RoundTripper, vim25.TemporaryNetworkError(int(connection.RoundTripperCount)))
+	minvcversion, err := semver.New("6.7.0")
+	if err != nil {
+		klog.Errorf("Failed to get parse vCenter version: 6.7.0. err: %+v", err)
+	} else {
+		vcversion, err := semver.New(client.ServiceContent.About.Version)
+		if err != nil {
+			klog.Errorf("Failed to parse vCenter version: %s. err: %+v", client.ServiceContent.About.Version, err)
+		} else {
+			result := vcversion.Compare(*minvcversion)
+			if result == -1 {
+				// vcversion is less than minvcversion
+				klog.Warningf("vSphere version: %s is deprecated. Please consider upgrading vSphere to 6.7u3 or higher", client.ServiceContent.About.ApiVersion)
+			} else if result == 0 {
+				// vcversion is equal to minvcversion
+				// check patch version
+				vcapiversion, err := semver.ParseTolerant(client.ServiceContent.About.ApiVersion)
+				if err != nil {
+					klog.Errorf("Failed to parse vCenter api version: %s. err: %+v", client.ServiceContent.About.ApiVersion, err)
+				}
+				if vcapiversion.Patch < 3 {
+					klog.Warningf("vSphere api version: %s is deprecated. Please consider upgrading vSphere to 6.7u3 or higher", client.ServiceContent.About.ApiVersion)
```

Comment:
I think, `vSphere patch version` instead of `vSphere api version` sounds better to me. Feel free to ignore this comment

Reply:
done